PetGame/Controllers/LeaderboardController.cs

```diff
@@ -5,85 +5,63 @@
 
 namespace PetGame
 {
+    public class LeaderboardEntry
+    {
+        public string PetName { get; set; }
+        public int score { get; set; }
+        public string OwnerName { get; set; }
+
+        public LeaderboardEntry(string nPetName, int nScore, string nOwnerName)
+        {
+            PetName = nPetName;
+            score = nScore;
+            OwnerName = nOwnerName;
+        }
+    }
+
     [Route("api/[controller]")]
     public class LeaderboardController : Controller
     {
         //initialize connection to SQL Server
         private readonly SqlManager sqlManager;
 
-        public LeaderboardController(SqlManager sqlManager) : base()
+        public LeaderboardController(SqlManager sqlManager)
         {
             this.sqlManager = sqlManager;
         }
 
         [HttpGet]
-        public IActionResult Get()
+        public IActionResult Get(int NumResults = 10)
         {
-            //create lists to hold the information
+            //create list to hold the information
             //names of pets on leaderboard
-            List<string> TopTenPetNames = new List<string>();
-            //scores of pets on leaderboard
-            List<Int32> TopTenScores = new List<Int32>();
-            //owner names of pets on leaderboard
-            List<string> TopTenOwnerNames = new List<string>();
-            //data structure to be returned as json
-            string[,] table = new string[3,10];
+            List<LeaderboardEntry> ScoreList = new List<LeaderboardEntry>();
             //Query db for races
             using (var conn = sqlManager.EstablishDataConnection)
             {
                 //create and set the SQL query
                 var cmd = conn.CreateCommand();
 
                 //to be optimized later, if time allows
-                cmd.CommandText = @"SELECT TOP (10) Pet.[Name] AS 'Pet Name', Race.Score, [User].Username AS 'Owner Name' 
-                                    FROM Pet, Race, [User] 
-                                    WHERE Race.PetId = Pet.PetId AND [User].UserId = Pet.UserId ORDER BY Score DESC;";
+                cmd.CommandText = @"SELECT TOP (" + NumResults + ") Pet.[Name] AS 'PetName', Race.Score, [User].Username AS 'OwnerName' " +
```

Comment:
Instead of string concatenation, use parameters (`cmd.Parameters.AddWithValue("@NumResults");`). Currently this is potentially SQLi vulnerable.